integration/api_endpoints_test.go

```diff
@@ -81,6 +81,7 @@ func TestConfigAPIEndpoint(t *testing.T) {
 	// Start again Mimir in single binary with the exported config
 	// and ensure it starts (pass the readiness probe).
 	require.NoError(t, writeFileToSharedDir(s, mimirConfigFile, body))
+	require.NoError(t, setDirPermission(s.SharedDir()))
```

Comment:
can we have `newMimirSingleBinaryWithLocalFilesytemBucket` set the permission on the SharedDir instead of doing it per test?

Reply:
I was thinking wrap it in a more general function. It should be nit. Then since we have very little tests that are actually request write permission, I thought it might be overkill to chmod on all tests. If this is a blocking command, I am ok to make it general.

Reply:
I see the function already existing, in the end we don't need to change tests anymore, seeing in backend enterprise docker image we never set USER, it was actually by default always run as root, when I remove the noroot user setup, the tests works now...